packages/core/src/lib/actions/callback/oauth/callback.ts

```diff
@@ -238,3 +243,43 @@ export async function getUserAndAccount(
     )
   }
 }
+
+async function getCryptoPrivateKeyFromOptions(
+  options: InternalOptions<"oauth" | "oidc">
+) {
+  return await getCryptoKeyFromPrivateKey(
+    getPrivateKeyValueFromOptions(options)
+  )
+}
+
+function getPrivateKeyValueFromOptions(
+  options: InternalOptions<"oauth" | "oidc">
+) {
+  if (
+    options.provider.client?.token_endpoint_auth_method === "private_key_jwt" &&
+    !options.provider.clientOptions?.clientPrivateKey
+  ) {
+    throw new TypeError("No private key found for private_key_jwt auth method.")
+  }
+
+  return options.provider.clientOptions!.clientPrivateKey!
+}
+
+async function getCryptoKeyFromPrivateKey(pem: string) {
+  const encodedKey = pem
+    .replace(/-----BEGIN PRIVATE KEY-----/, "")
+    .replace(/-----END PRIVATE KEY-----/, "")
+    .replace(/\n/g, "")
+  const encodedKeyBuffer = Buffer.from(encodedKey, "base64")
+
+  return await crypto.subtle.importKey(
+    "pkcs8",
+    encodedKeyBuffer,
+    {
+      name: "RSASSA-PKCS1-v1_5", // algorithm
+      hash: "SHA-256", // hash algorithm
+    },
+    true,
+    ["sign"]
+  )
+}
```

Comment:
```suggestion
```

I think this should live in user-space for now. `Buffer` is also Node.js specific, and we want to keep `@auth/core` to only use Web APIs, so it would need a slightly different implementation.

Reply:
Yes, fair point 👍